### DISCUSSION TITLE
Cannot use \includesvg{} and \input{} commands inside markdown anymore

### DISCUSSION THREAD

**Discussion by sinaatalay** [Witiko/markdown#186] [Q&A] [answered]
This package is so excellent that it made me love writing. I have been using it for more than eight months now. However, after I updated to TeX Live 2022, everything suddenly stopped working. After some tinkering, I found out what was going on.

This is the usual setup that I have always been using:

```
\documentclass[]{article}

\usepackage{tikz} % To draw figures
\usepackage{pgfplots} % To create plots
\usepackage{svg} % To include svg images

\usepackage{markdown}

\markdownSetup{
  hybrid,
  rendererPrototypes={
    image = {
      \begin{figure}[ht!]
        \centering
        #3
        \caption{#4}
        \label{fig:#1}
      \end{figure}
    }
  }
}
```
I am using this setup because sometimes, instead of `\includegraphics{}`, I may include SVG figures or tikz figures. Like these:

```
\begin{document}

  \begin{markdown}
    ![exampleLabel1](\includegraphics{exampleImage.png) "Example caption 1.")
    
    ![exampleLabel2](\includesvg{exampleSVG.svg) "Example caption 2.")
    
    ![exampleLabel3](\input{myfigure.tikz) "Example caption 3.")
  \end{markdown}
  
\end{document}
```

Unfortunately, markdown package cannot render these anymore. It shows the TeX comments inside `myfigure.tikz` as normal text and fails to render the figure. Moreover, `\includesvg{exampleSVG.svg)` doesn't seem to work at all.

When I select TeX Live 2021 on Overleaf, everything works fine. But TeX Live 2022 fails. I think it is a big issue. However, no one seems to notice it yet.

**Comment by Witiko** [comment, marked as answer]
Dear @sinaatalay, glad to hear that the Markdown package has been so enjoyable to you and sorry to hear about your trouble. Since the conversion from markdown to TeX does not obey lines, TeX comments commonly consumed more input than intended, which was unintuitive and a security issue. Since version 2.10.0, percent signs (`%`) are treated as letters with the `hybrid` option, see also https://github.com/Witiko/markdown/issues/42 and https://github.com/Witiko/markdown/issues/89. Since 2.16.1, the same is true of hash signs (`#`).

Here is how you can change your code to take this change into account. Your code should still work with TeX Live 2021 after the change:

``` tex
\markdownSetup{
  hybrid,
  rendererPrototypes={
    image = {
      \begin{figure}[ht!]
        \centering
        \catcode`\%=14\relax
        \catcode`\#=6\relax
        #3
        \catcode`\%=12\relax
        \catcode`\#=12\relax
        \caption{#4}
        \label{fig:#1}
      \end{figure}
    }
  }
}
```

**Reply by sinaatalay** [reply]
Thank you so much! Now, everything works fine. Thanks for making the LaTeX experience more maintainable and easier with this package 🥇 .
